src/web_clipboard.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::{
+    mpsc::{self, Receiver, Sender},
+    Arc, Mutex,
+};
+
+use bevy::prelude::*;
+use wasm_bindgen_futures::spawn_local;
+
+use crate::EguiClipboard;
+
+/// startup system for bevy to initialize clipboard.
+pub fn startup(mut clipboard_channel: ResMut<EguiClipboard>) {
+    setup_clipboard_paste(&mut clipboard_channel.clipboard)
+}
+
+fn setup_clipboard_paste(clipboard_channel: &mut WebClipboardPaste) {
+    let (tx, rx): (Sender<String>, Receiver<String>) = mpsc::channel();
+
+    use wasm_bindgen::closure::Closure;
+    use wasm_bindgen::prelude::*;
+
+    let closure = Closure::<dyn FnMut(_)>::new(move |event: web_sys::ClipboardEvent| {
+        // TODO: maybe we should check if current canvas is selected ? not sure it's possible,
+        // but reacting to event at the document level will lead to problems if multiple games are on the same page.
```

Comment:
It should be pretty much possible, yeah. I'm ok if this isn't covered in this PR, but I guess the way to tackle it would look as follows:
1. Access a `winit` window via https://docs.rs/bevy/0.10.1/bevy/winit/struct.WinitWindows.html
2. Get the related canvas element: https://docs.rs/winit/latest/wasm32-unknown-unknown/winit/platform/web/trait.WindowExtWebSys.html#tymethod.canvas (it can be referenced as `web_sys::Element` too: https://docs.rs/web-sys/latest/web_sys/struct.HtmlCanvasElement.html#impl-AsRef%3CElement%3E-for-HtmlCanvasElement)
3. Compare it with the document's https://docs.rs/web-sys/latest/web_sys/struct.Document.html#method.active_element

Reply:
Created a follow up issue (a bit early but hopefully we'll manage to merge this PR)

https://github.com/mvlabat/bevy_egui/issues/185